Add Playable Colony

SierraKomodo opened the pull request:
Another @Aticius PR.

:cl: Aticius
add: Adds a joinable Colony map that can spawn on exoplanets. Rimworld mode is here.
/:cl:

Aticius commented:
=
![dreammaker_2018-11-23_17-18-08](https://user-images.githubusercontent.com/7741012/48966519-bf65fe00-ef98-11e8-9a5e-7a6e1fac7a26.png)
![dreammaker_2018-11-23_17-18-21](https://user-images.githubusercontent.com/7741012/48966520-c12fc180-ef98-11e8-9da3-e97d32fff20a.png)

Preview images for ease.

Filling the rest out for Sierra:

Adds a 6-slot, Joinable, Exoplanet Ruin meant as a fully self-sustaining Colony. Equipment is kept to reliable things where possible.

Intended to be pretty chill, all things considered, to contrast the trend of "NEED HELP NOW!!" away sites.

Things of note that probably need discussing:

Armory is pretty well stocked, mostly to dissuade dunga explorers from thinking this place is easy lootin' and dealing with the inevitable pain and torture that is Spider Planets. 
Merchant Pad exists, but Colony has access to, on average, less than 3000~ Thalers of cash to spend.
Lots of high end stock parts, intended to be used to build some fusion generators to help give Colonists something to do.

Miraviel commented:
Make sure to include a changelog!

Alex6511 commented:
Petition to make this a lowpop map to replace the Torch. (In addition to being off-ship during highpop)

CakeQ commented:
How many slots does it have for players?

Aticius commented:
> How many slots does it have for players?

6, As stated.

CakeQ commented:
I'm blind

SierraKomodo commented:
Changelog added

afterthought2 commented:
Requires the crucial cow fix from #23845, it seems.

Aticius commented:
Added @afterthought2's requested changes.

afterthought2 reviewed (dismissed):
I think this is OK, but it needs that cow fix before merging lest you want random Travis fails. We really should make a unit test that spawns all ruins, btw.

Ithalan commented:
I would suggest widening the medbay by 1 tile so there's a room to maneuver a rollerbed around in there, just for the sanity of whoever has to actually use the place somewhere down the line. Mapping in a rollerbed and perhaps a stasis/body bag or two in the process also seems appropriate for something of this size.

I would also add a sink to the medbay, both for washing hands prior to surgery and for emptying beakers from dialysis/chemistry.

In the absence of a disposals system, maybe add in some trash carts/bins around the place so trash can at least easily be collect for disposal in the crematorium in the courtyard?

babydoll commented:
Did you test to ensure that the grass tiles properly override the planet's atmosphere? That was an issue with another exomap ruin previously.

Double-wide full windows look pretty bad on the compiled map.

Aticius commented:
> Did you test to ensure that the grass tiles properly override the planet's atmosphere? That was an issue with another exomap ruin previously.
> 
> Double-wide full windows look pretty bad on the compiled map.

All of the playtesting done on the compiled map hasn't shown any problems with the animal pen and the corresponding grass tiles.

And the double wides seem fine to me!

SierraKomodo committed:
Add Playable Colony
mapmerge
Fixes snow planet ore maps

They were so poor they didn't pass sanity. Now sanity is more tweakable.

Also fixes lava planet's ultra rich ore map actually being super poor.

SierraKomodo commented:
Aticius made some changes adding intercoms and reworking some other stuff

CakeQ commented:
- Colonists get assault rifle
- Colonists get anti-materiel rifle
- Colonists get an e-gun to shoot through windows with
- Colonists get an electrified grille perimiter wall
- Colonists get reinforced phoron windows
- Colonists get fully reinforced exterior walls
- Colonists get merchant computer to trade for infinite supplies
- Colonists have a better life than people on the Torch

This away site is not designed to be interacted with at all by explorers, nor does it have any potential danger or risk to the colonists, and can only be described as a hugbox.

Not really sure why this has been approved as something the the explorers are meant to explore.

Aticius commented:
> * Colonists get assault rifle
> * Colonists get anti-materiel rifle
> * Colonists get an e-gun to shoot through windows with
> * Colonists get an electrified grille perimiter wall
> * Colonists get reinforced phoron windows
> * Colonists get fully reinforced exterior walls
> * Colonists get merchant computer to trade for infinite supplies
> * Colonists have a better life than people on the Torch
> 
> This away site is not designed to be interacted with at all by explorers, nor does it have any potential danger or risk to the colonists, and can only be described as a hugbox.
> 
> Not really sure why this has been approved as something the the explorers are meant to explore.

Most of this was covered in #feedback, i'll try my best to paraphrase it.

Assault Rifles: 
The pair of STSes is an flavor choice to tie in for them being Frontiersy guns and, unless i'm misremembering, are the least robust choice of gun in the armory if not firing in bursts, which would counteract the main selling point of it.

Anti-Material Rifle:
It isn't. It's an Ant Rifle. It's 9mm. It's a joke. Has minor utility in sniping animals if you can spot them before they spot you.

E-Guns: Small Eguns are pretty shit against a majority of exoplanet mobs, but they're useful.

Guns, overall:

Who are they going to shoot, exactly? Themselves? The Wildlife? Explorers trying to steal their stuff? Colony exists a bit out of the balance loop, given that Exoplanets can range from benign and wonderful to literally filled with millions of spiders. Need to give them something to defend themselves with.

Electrified Grille: 

Mostly serves to harm the Colonists more than the animals. Also the Explorers.

Borosilicate Windows: Normal R-Windows would be at risk of breaking from volcanic worlds. I felt that making a special snowflake window just for this map would be a bit out of scope. I can't think of anything really inherently wrong with them, given that mobs still break through and they'd just get E-gunned anyways. 

R-Walls: Same deal here. Steel would be liable to melt and would make breaking into the colony far too easy. Still fail to see how they're inherently bad.

Merchant Computer:

This gives them something to work towards if they're ignored. Their starting funds are, on average, 2/3rd that of the merchant, without an easy way to get more.

Selling stuff is still broken, anyhow. This is a bit hyperbole given how buggy and poorly maintained trading is.

Better Life: 

Damn right. Start a new one.

As for Danger:

There's nothing INHERENTLY endangering to the Colony at large. the building is safe. You can sit inside, watch power carefully and do nothing, if you want. Or you can go outside and expose yourself to the myriad dangers of whatever hell hole you're stuck on. The Playtesting done shows that the airlock is VERY much not fast enough for a dying person to get through without help, save for a miracle.

Play safe, stay safe. Play risky, stay risky. It's your choice.

Things will be adjusted as more people play.

comma commented:
>Not really sure why this has been approved as something the the explorers are meant to explore.

I am pretty sure those guns are exactly there so explorers don't explore it without permission.
This isn't something explorers are meant to explore, but NPCs for explorers to chat with :^)
